### PR TITLE
Federated Identity take 2

### PR DESCRIPTION
**Pull request by rwz**
In this case we introduce a Finalize endpoint that can receive a signed token and write cookies based on it. This will be used by Fido in federated identity workflow.

### CODE REVIEW

**Comment by dmcinnes** [comment]
:+1:

File: lib/identity/finalize.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+module Identity
+  class Finalize < Default
```
Comment (brandur) [inline lib/identity/finalize.rb]: I know that the existing convention is a bit light, but can you add an explanatory comment indicating what this module is for?

File: lib/identity/finalize.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    register Sinatra::Namespace
+    include Helpers::Auth
+
+
+    namespace "/finalize" do
```
Comment (brandur) [inline lib/identity/finalize.rb]: I'm not super in love with this URL seeing as it doesn't really describe what this endpoint does. Did you guys consider any alternatives? I wonder if we should try something like `/login/external`, `/login/with-token`, `/login/with-authorization` or something?

File: lib/identity/finalize.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
+module Identity
+  class Finalize < Default
+    register Sinatra::Namespace
+    include Helpers::Auth
+
+
```
Comment (brandur) [inline lib/identity/finalize.rb]: Mind just trimming your whitespace here? Thanks!

File: lib/identity/config.rb [hunk 4]
```diff
@@ -2,6 +2,10 @@ module Identity
   module Config
     extend self
 
+    def finalize_shared_secret
```
Comment (brandur) [inline lib/identity/config.rb]: Hey, this file sort of fell into disrepair, but it's still mostly alphabetized. Mind just sorting this between "c" and "h"? Thanks!

**Comment by brandur** [comment]
+1 guys! Looks awesome.

**Comment by dmcinnes** [comment]
Pulling this in

**Pull request merged**